src/workflows/server/server.py

```diff
@@ -0,0 +1,220 @@
+    #
+
+    async def _health_check(self, request: Request) -> JSONResponse:
+        return JSONResponse({"status": "healthy"})
+
+    async def _list_workflows(self, request: Request) -> JSONResponse:
+        workflow_names = list(self._workflows.keys())
+        return JSONResponse({"workflows": workflow_names})
+
+    async def _run_workflow(self, request: Request) -> JSONResponse:
+        workflow = self._extract_workflow(request)
+        context, start_event, run_kwargs = await self._extract_run_params(
+            request, workflow
+        )
+
+        try:
+            result = await workflow.run(
+                ctx=context, start_event=start_event, **run_kwargs
+            )
+            return JSONResponse({"result": result})
+        except Exception as e:
+            raise HTTPException(detail=f"Error running workflow: {e}", status_code=500)
+
+    async def _run_workflow_nowait(self, request: Request) -> JSONResponse:
+        workflow = self._extract_workflow(request)
+        context, start_event, run_kwargs = await self._extract_run_params(
+            request, workflow
+        )
+        handler_id = nanoid()
+
+        self._handlers[handler_id] = workflow.run(
+            ctx=context, start_event=start_event, **run_kwargs
+        )
+        return JSONResponse({"handler_id": handler_id, "status": "started"})
+
+    async def _get_workflow_result(self, request: Request) -> JSONResponse:
+        handler_id = request.path_params["handler_id"]
+        handler = self._handlers.pop(handler_id, None)
+        if handler is None:
+            raise HTTPException(detail="Handler not found", status_code=404)
+
+        try:
+            result = await handler
+
+            return JSONResponse({"result": result})
+        except Exception as e:
+            return JSONResponse({"error": str(e)}, status_code=500)
+
+    async def _stream_events(self, request: Request) -> StreamingResponse:
+        handler_id = request.path_params["handler_id"]
+        handler = self._handlers.get(handler_id)
+        if handler is None:
+            raise HTTPException(detail="Handler not found", status_code=404)
+
+        # Get raw_event query parameter
+        sse = request.query_params.get("sse", "false").lower() == "true"
+        media_type = "text/event-stream" if sse else "application/x-ndjson"
+
+        async def event_stream(handler: WorkflowHandler) -> AsyncGenerator[str, None]:
+            serializer = JsonSerializer()
+
+            async for event in handler.stream_events():
+                serialized_event = serializer.serialize(event)
+                if sse:
+                    # need to convert back to str to use SSE
+                    event_dict = json.loads(serialized_event)
+                    yield f"event: {event_dict.get('qualified_name')}\ndata: {json.dumps(event_dict.get('value'))}\n"
+                else:
+                    yield f"{serialized_event}\n"
+
+                await asyncio.sleep(0.01)
```

Comment:
nit: I think sleeping 0 should usually be enough to allow the next tasks in line in the event loop to run
```suggestion
                await asyncio.sleep(0)
```